### PR TITLE
Enable support for multiple JWT audiences

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**

### CODE REVIEW

**Comment by msrb** [comment]
[test this]

**Comment by samuzzal-choudhury** [comment]
Please do not merge the PR.

**Comment by samuzzal-choudhury** [comment]
Ready to be reviewed and merged

File: bayesian/auth.py [hunk 1]
```diff
@@ -24,12 +24,23 @@ def decode_token():
...
+        try:
+            decoded_token = jwt.decode(token, pub_key, audience=aud)
+        except jwt.InvalidTokenError:
+            current_app.logger.error('Auth Token could not be decoded for audience {}'.format(aud))
+            decoded_token = None
+
+        if decoded_token == None:
+            continue
+        else:
+            break
```
Comment (fridex) [inline bayesian/auth.py]: Keeping only:

```python
if decoded_token is not None:
   break
```

should do the trick.

File: bayesian/auth.py [hunk 2]
```diff
@@ -24,12 +24,23 @@ def decode_token():
...
+
+    if decoded_token == None:
+        raise jwt.InvalidTokenError
```
Comment (fridex) [inline bayesian/auth.py]: Probably some descriptive error message for logs - instantiate exception class.

**Comment by samuzzal-choudhury** [comment]
@fridex if you may recheck :-)

**Comment by samuzzal-choudhury** [comment]
@fridex thanks for approving.

**Pull request merged**